session1/solutions/assignment4/Rakefile

```diff
@@ -1 +1,21 @@
-# assignment 4 solution
+require 'rake/packagetask'
+
+desc 'Archive a folder'
+task :archive do
+  path = ENV["MYPATH"] || Dir.pwd
+  exclude = ENV["EXCLUDE"]
+  archive_name = path.split("/").last
+  archive_name.gsub!(/\W/, '_')
+  filelist = FileList.new("#{path}/*")
+  unless exclude.nil?
```

Comment:
Code is short no doubt - which is good. You should consider adding spaces before and after this block and in places where there are different code procedures like after parameter declarations (line 9) and after blocks (line 15)